tests/data_context/test_data_context_datasource_non_sql_methods.py

```diff
@@ -316,3 +316,192 @@ def my_custom_batch_selector(batch_identifiers: dict) -> bool:
         "name": "Test",
         "year": "2010",
     }
+
+
+def test_get_batch_list_from_new_style_datasource_with_file_system_datasource_configured_assets_testing_limit_and_custom_filter(
+    empty_data_context, tmp_path_factory
+):
+    context = empty_data_context
+    base_directory = str(
+        tmp_path_factory.mktemp(
+            "test_get_batch_list_from_new_style_datasource_with_file_system_datasource_configured_assets_testing_limit_and_custom_filter"
+        )
+    )
+    create_files_in_directory(
+        directory=base_directory,
+        file_name_list=[
+            "Test_1998.csv",
+            "Test_1999.csv",
+            "Test_2000.csv",
+            "Test_2010.csv",
+            "Test_2021.csv",
+        ],
+        file_content_fn=lambda: "x,y,z\n1,2,3\n2,3,5",
+    )
+
+    config = yaml.load(
+        f"""
+    class_name: Datasource
+
+    execution_engine:
+        class_name: PandasExecutionEngine
+
+    data_connectors:
+        my_data_connector:
+            class_name: ConfiguredAssetFilesystemDataConnector
+            base_directory: {base_directory}
+            glob_directive: "*.csv"
+
+            default_regex:
+                pattern: (.+)_(\\d.*)\\.csv
+                group_names:
+                    - name
+                    - year
+            sorters:
+                - orderby: desc
+                  class_name: NumericSorter
+                  name: year
+            assets:
+                YearTest:
+                    base_directory: {base_directory}
+                    pattern: (.+)_(\\d.*)\\.csv
+                    group_names:
+                        - name
+                        - year
+
+        """,
+    )
+    context.add_datasource(
+        "my_datasource",
+        **config,
+    )
+
+    # only select files from 1999 or later
+    def my_custom_batch_selector(batch_identifiers: dict) -> bool:
+        return int(batch_identifiers["year"]) >= 1999
+
+    # Use an instantiated BatchRequest object with custom filter AND limit
+    batch_request: BatchRequest = BatchRequest(
+        datasource_name="my_datasource",
+        data_connector_name="my_data_connector",
+        data_asset_name="YearTest",
+        data_connector_query={"custom_filter_function": my_custom_batch_selector},
+        limit=2,
```

Comment:
@anthonyburdi Do we need a test to see if the presence of the `limit` key in `batch_spec_passthrough` will override the existing values thereof?  Thanks!

Reply:
Good call, adding!